libnopegl/src/node_drawother.c

```diff
@@ -812,9 +812,18 @@ static int finalize_pipeline(struct ngl_node *node,
     if (ret < 0)
         return ret;
 
+    ngli_darray_init(&desc->textures_map, sizeof(struct texture_map), 0);
+    const struct pgcraft_compat_info *info = ngli_pgcraft_get_compat_info(desc->crafter);
+    for (size_t i = 0; i < info->nb_texture_infos; i++) {
+        const struct texture_map map = {.image = info->images[i], .image_rev = SIZE_MAX};
+        if (!ngli_darray_push(&desc->textures_map, &map))
+            return NGL_ERROR_MEMORY;
+    }
+
```

Comment:
Can you put it in a dedicated function: `build_textures_map` ?

Reply:
Good idea, done: I also moved the initialization of the `textures_map` along with the others